ios/Classes/AvailabilityEventStreamHandler.swift

```diff
@@ -0,0 +1,34 @@
+import HyperTrack
+
+class AvailabilityEventStreamHandler: NSObject, FlutterStreamHandler {
+    
+    private var eventSink: FlutterEventSink?
+    
+    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
+        eventSink = events
+        NotificationCenter.default.addObserver(self, selector: #selector(onAvailable), name: HyperTrack.becameAvailableNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(onUnavailable), name: HyperTrack.becameUnavailableNotification, object: nil)
+        
+        HyperTrackSDKWrapper.withSdkInstance { (sdk: HyperTrack) in
+            eventSink!(serializeIsAvailable(sdk.availability))
+            return .success(.void)
+        }.sendErrorIfAny(eventSink!, errorCode: errorCodeStreamInit)
+        
+        return nil
+    }
+    
+    func onCancel(withArguments arguments: Any?) -> FlutterError? {
```

Comment:
you should remove the `NotificationCenter` observer here, otherwise when a new notification is received the `eventSink!` accessor will crash the app